hw05.sh

```diff
@@ -0,0 +1,40 @@
+##usage bash hw05.sh wages.csv <column designation of gender> <column designation of yearsExperience> <column designation of wage>
```

Comment:
1.	Question is looking for the uniq combinations of gender and years experience and it wants these written to a file. Your code works, in order to fully answer the question you’d need to add uniq to your pipeline and write stdout to > gender_experience.csv
a.	0.8
2.	Code works but produces the wrong answers. It’s output is a male with 9 years experience and a wage of 9; female with 1 year experience and wage of 12; 9 females in top 10 earners. Correct answers area    
gender	yearsExperience	yearsSchool	wage
male	5	11	39.80892
female	9	11	0.076556
Code that would produce this: 
cat wages.csv| grep -v 'gender' | sort -t , -k 4 -n -r | head -n 1 | cut -d , -f 1,2,4
cat wages.csv | grep -v "gender" | sort -t , -k 4 –n | head -n 1 | cut -d , -f 1,2,4
cat wages.csv | grep -v "gender" | sort -t , -k 4 -n -r | head -n 10 | grep "female" | cut -d , -f 1,2
a.	0.7
3.	Code works but doesn’t quite get to the right answer. It says that the difference in minimum wage for people with a college degree vs. high school degree is only 5 cents. This comes from a slight error in your code. When you grep 12 it will pull rows with 12 years experience or 12 years school or wages that have 12 in them since grep will consider partial matches as matches. One way around this is to cut to just the school and wage column and grep for 12 or 16 with a , after it in order to match the school column and not a partial match to a wage. Here’s some code that would do that: cat wages.csv | cut -d , -f 3,4 | grep '16,' |  sort -t , -k 2 -n | head -n 1 | cut -d , -f 2
a.	0.7

2.2/3